enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+three different deployment models: 
+
+1. **Manual Deployment** of a `bpfProgramConfig` by a cluster administrator with
+   no accompanying application. 
+2. **Manual Deployment** of a `bpfProgramConfig` by a cluster administrator with
+   an accompanying application. 
+3. **Dynamic Deployment** of a `bpfProgramConfig` where the application deploys
+   and manages the BPF program while also interacting with it on a per-node
+   basis. 
+
+For all deployment scenarios, the application or a portion of it needs to be
+deployed as a per-node agent with a mount-point linked to the bpfd-fs
+`/run/bpfd/fs/`. Each application agent will then use the bpfd-helpers library
+(still WIP) to fetch the node-specific `bpfProgram` which is used internally to
+determine the absolute pin path of the maps on the node.
+
+For Deployment scenario 3, the application would need to have a centralized
+operator. In this case, the operator will use the bpfd-helpers library to manage
+the `bpfProgramConfig` object and dynamically configure fields such as bpfd
+bytecode location, attach points, priority (when applicable), direction, etc.
+
+### API Extensions
+
+The bpfd operator uses the following APIs:
+
+#### BpfProgramConfig CRD
+
+The [BpfProgramConfig
+CRD](https://github.com/redhat-et/bpfd/blob/main/bpfd-operator/config/crd/bases/bpfd.io_bpfprogramconfigs.yaml)
+is the bpfd K8s API object most relevant to users and can be used to understand
+clusterwide state for a BPF program. It's designed to express how, and where BPF
+programs are to be deployed within a Kubernetes cluster. An example
+BpfProgramConfig which loads a basic xdp-pass program to all nodes can be seen
+below:
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+kind: BpfProgramConfig
+metadata:
+  labels:
+    app.kubernetes.io/name: BpfProgramConfig
+  name: xdp-pass-all-nodes
+spec:
+  ## Must correspond to image section name
+  name: pass
+  type: XDP
+  # Select all nodes
+  nodeselector: {}
+  priority: 0
+  attachpoint: 
+    interface: eth0
+  bytecode:
+    imageurl: quay.io/bpfd-bytecode/xdp_pass:latest
+```
+
+
+#### BpfProgram CRD
+
+The [BpfProgram
+CRD](https://github.com/redhat-et/bpfd/blob/main/bpfd-operator/config/crd/bases/bpfd.io_bpfprograms.yaml)
+is used internally by the bpfd-deployment to keep track of per node bpf program
+state such as program UUIDs and map pin points, and to report node specific
+errors back to the user. K8s users/controllers are only allowed to view these
+objects, NOT create or edit them. Below is an example BpfProgram Object which
+was automatically generated in response to the above BpfProgramConfig Object.
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+  kind: BpfProgram
+  metadata:
+    creationTimestamp: "2022-12-07T22:41:29Z"
+    finalizers:
+    - bpfd.io.agent/finalizer
+    generation: 2
+    labels:
+      owningConfig: xdp-pass-all-nodes
+    name: xdp-pass-all-nodes-bpfd-deployment-worker2
```

Comment:
How does an agent running on a given node know which programs to pay attention to?